source/neuropod/bindings/java/src/main/java/com/uber/neuropod/LibraryLoader.java

```diff
@@ -1,19 +1,143 @@
+                                                                "libneuropod_tensorflow_backend.so",
+                                                                "libneuropod_torchscript_backend.so",
+                                                                "libneuropod_pythonbridge_backend.so",
+                                                                // TensorFlow libraries name
+                                                                "libtensorflow_framework.so",
+                                                                "libtensorflow.so",
+                                                                // TorchScript libraries linux name
+                                                                "libc10.so",
+                                                                "libcaffe2_detectron_ops.so",
+                                                                "libcaffe2_module_test_dynamic.so",
+                                                                "libcaffe2_observers.so",
+                                                                "libfbjni.so",
+                                                                "libgomp-7c85b1e2.so.1",
+                                                                "libpytorch_jni.so",
+                                                                "libshm.so",
+                                                                "libtorch_cpu.so",
+                                                                "libtorch_global_deps.so",
+                                                                "libtorch_python.so",
+                                                                "libtorch.so",
+                                                                // TorchScript libraries darwin name
+                                                                "libc10.dylib",
+                                                                "libcaffe2.dylib",
+                                                                "libmklml.dylib",
+                                                                "libtorch.1.dylib",
+                                                                "libtorch.dylib",
+                                                                "libiomp5.dylib"
+                                                                );
+    private static final List<String> BIN_NAMES = Arrays.asList("neuropod_multiprocess_worker");
+    private static final String JNI_NAME = "neuropod_jni";
+    private static final String JNI_EMBEDDED_NAME = "libneuropod_jni.so";
+    private static final int BUFFER_SIZE = 1<<20;
+    private static final String TEMP_DIRECTORY_NAME = "neuropod_native_libraries";
+
     /**
      * Load jni library.
      */
     public static void load() {
+        if (isLoaded() || loadSharedLibrary() || loadEmbeddedLibrary()) {
```

Comment:
Interesting, is there guarantee that this is called w/o concurrency and so no multiple therads? Otherwise it will need some locking/memory barrier to guarantee correctness. (2 callers may get isLoaded() == false at the same time and proceed with calling load* methods)

If this is guaranteed, please specify in the comment of method that there is a guarantee on it.

Reply:
Yes, Java static initializers are thread safe